Makefile

```diff
@@ -2,6 +2,9 @@
 
 OCAML_VERSION_RECOMMENDED=4.06.1
 
+TMP=${TMPDIR:-/tmp}
```

Comment:
The `:-`  allows for a variable to be assigned a value if another variable is either empty or is undefined (according to 
https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html)

Reply:
Can you add the comment about `:-` in the Makefile also? I'm not very sure that we should do this, because the blockchain code actually would just use `/tmp/zilliqa.sock`.

Reply:
> the blockchain code actually would just use `/tmp/zilliqa.sock`

Yeah, I'm not 100% sure about it.
This change is intended mostly for non-FHS compliant operating systems (but SUS-compliant at the same time): GoboLinux and NixOS are examples of intentionally non-compliant filesystem implementations. NixOS actually has the `/tmp` directory. The reason I've proposed this is that I had some issues with using the `/tmp` for storing the `zilliqa.sock` file (when running scilla testsuite from the `nix-shell`). But it should be changed accordingly in the `constants.xml` or `constants_local.xml` as well... So maybe it doesn't make much sense.

Reply:
> But to make it work the path to IPC socket file should be changed accordingly in the constants.xml or constants_local.xml as well... So maybe it doesn't make much sense.

Exactly, and it is very unlikely that the blockchain codebase will bother about non Ubuntu OSes in the near future. Let's drop this part and just have the other part (since the one in the testsuite is internal to Scilla).

Reply:
At best, we can have it as a variable in the `Makefile`. `SOCK_PATH=/tmp/zilliqa.sock` and use that variable subsequently.